File: code/2026/Lokal_LLM_for_Saksbehandlere/WindowsApplication/LocalLLMApp/Models/ApiModels/APIData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

// <Purpose>
// A general structure for the data we want to receive from API calls
// </Purpose>

namespace LocalLLMApp.Models.ApiModels
{
    public class APIData
    {
        public User? User { get; set; } = new();
        public List<Case>? Cases { get; set; } = new();
        public List<Chat>? Chats { get; set; } = new();
        public List<ChatMessage>? Messages { get; set; } = new();
        public List<FilesInfo>? Files { get; set; } = new();
    }
}

File: code/2026/Lokal_LLM_for_Saksbehandlere/Backend/classes/apidata.py
from pydantic import BaseModel
from .user import UserData
from .case import Case
from .chat import Chat
from .chatmessage import ChatMessage
from .filesinfo import FilesInfo


# Data model for receiving and sending APIData to the .NET WinUI app
class APIData(BaseModel):
    User: UserData
    Cases: list[Case] = []
    Chats: list[Chat] = []
    Messages: list[ChatMessage] = []
    Files: list[FilesInfo] = []
    